iAM-legacy/Simulation/Investments.cs

```diff
@@ -321,104 +353,96 @@ public string IsBudgetAvailable(float fAmount, String strBudget, String strYear,
+                        throw e;
+                    }
+
+                    
+                    //This gets a hash of all years and the amount of budgets originally available.                    
+                    try
+                    {
+                        hashYearAmountOriginal = (Hashtable)BudgetYearOriginal[budgetCheck];
+                        //This occurs when a split treament goes pas then end.
+                        if (!hashYearAmountOriginal.ContainsKey(currentYear.ToString())) continue;
+                    }
+                    catch (Exception e)
+                    {
+                        SimulationMessaging.AddMessage(new SimulationMessage("Error: Retrieving budget year original. " + e.Message));
+                        throw e;
+                    }
+
+                    
+                    try
+                    {
+                        if (!hashYearAmountOriginal.Contains(strYear)) continue;
+                        fOriginal = (float)hashYearAmountOriginal[strYear];
+                    }
+                    catch (Exception e)
+                    {
+                        SimulationMessaging.AddMessage(new SimulationMessage("Error: Retrieving year amount original. " + e.Message));
+                        throw e;
+                    }
+
+                    fAfterSpending = fAvailable - fAmount;
+                    if (fOriginal <= 0) continue;
+                    fPercent = (1 - fAfterSpending / fOriginal) * 100;
+
+
+                    try
                     {
                         float fPercentLimit = (float)priority.BudgetPercent[budgetCheck];
                         var original = (float)priority.BudgetPercent[budgetCheck] * fOriginal;
                         budgetHash = fAmount + "/" + fAvailable.ToString("f0") + "/" + original.ToString("f0");
-                        if (fPercent < fPercentLimit) return budgetCheck;
-                        
+                        //if (fPercent < fPercentLimit) return budgetCheck;
+                        if (fPercent > fPercentLimit) currentBudgetAvailable = false;
                     }
-                    else
+                    catch (Exception e)
                     {
-                        if (priority.Criteria.IsCriteriaMet(hashAttributeValue))
-                        {
-                            float fPercentLimit = (float)priority.BudgetPercent[budgetCheck];
-                            var original = (float)priority.BudgetPercent[budgetCheck] * fOriginal;
-                            budgetHash = fAmount + "/" + fAvailable.ToString("f0") + "(" + original.ToString("f2") + ")";
-                            if (fPercent < fPercentLimit) return budgetCheck;
-                        }
+                        SimulationMessaging.AddMessage(new SimulationMessage("Error: Evaluating priority. " + e.Message));
```

Comment:
`SimulationMessaging` is unusable in the web app. We need to have this written out to an error log file that we can get from PennDOT IT.

Reply:
Not a useful message for the web application.  More for debugging on the desktop.  Will properly wrap messages for the web.

Reply:
Change the message to be a useful log message and wrap in the WebAPI logging call.